nf_core/pipeline-template/lib/NfcoreSchema.groovy

```diff
@@ -264,6 +264,30 @@ class NfcoreSchema {
 
         // Get pipeline parameters defined in JSON Schema
         def Map params_summary = [:]
+        def LinkedHashMap params_mqc = [:]
+        params_mqc['parent_id'] = 'nf-core-{{short_name}}_run_details'
+        params_mqc['parent_name'] = '{{name}} Run Details'
+        params_mqc['id'] = 'nextflow_params'
+        params_mqc['section_name'] = 'Pipeline Parameters'
+        params_mqc['description'] = 'Lists all pipeline parameters used to control how the pipeline was run, including those not modified from the defaults.'
+        params_mqc['plot_type'] = 'table'
+        params_mqc['pconfig'] = [:]
+        params_mqc['pconfig']['id'] = 'nextflow_params_table'
+        params_mqc['pconfig']['title'] = 'Nextflow Pipeline Parameters'
+        params_mqc['pconfig']['namespace'] = 'Nextflow Pipeline'
+        params_mqc['headers'] = [:]
+        params_mqc['headers']['value'] = [:]
+        params_mqc['headers']['value']['title'] = 'Value'
+        params_mqc['headers']['value']['description'] = 'Defined value in the pipeline run'
+        params_mqc['headers']['value']['scale'] = false
+        params_mqc['headers']['value']['format'] = '{}'
+        params_mqc['headers']['default'] = [:]
+        params_mqc['headers']['default']['title'] = 'Default'
+        params_mqc['headers']['default']['description'] = 'Default value in the pipeline'
+        params_mqc['headers']['default']['scale'] = false
+        params_mqc['headers']['default']['format'] = '{}'
+        params_mqc['data'] = [:]
```

Comment:
🚀 Be nice to split the MultiQC component into a separate function if possible. This function is getting a little bloated!

Reply:
Yes. Which in fact is what had already been done in `WorkflowPipeline.groovy` with `paramsSummaryMultiqc()` 🤦🏻 

I thought the same but didn't want to spend too much more time on this PR before we agreed whether we actually wanted the functionality or not. If we do, then I can move it around and tidy it up a bit.

Reply:
I don't see an issue with having this as part of the pipeline template. Having a stand-alone module would make things more flexible in terms of changing things but having it in the template will make the implementation cleaner and hopefully a little more flexible?

Reply:
Agreed. With this much data, I would have put it in a YAML file and loaded it in with
```
import org.yaml.snakeyaml.Yaml 

def readYAML( yamlfile ) {
    return new Yaml().load( new FileReader( yamlfile.toString() ) )
}
```
But it's likely the JSONBuilder is the only class needed (just quick thoughts - still wondering if I understand).

Reply:
What about overlap with the existing summary section?

Reply:
We can actually be clever and stick all of these MultiQC config blocks into the bundled assets file. Just need them to have the `id` attribute so that we can reference them.